src/NClient.CodeGeneration/NClient.CodeGeneration.Abstractions/INClientGenerator.cs

```diff
@@ -0,0 +1,10 @@
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace NClient.CodeGeneration.Abstractions
+{
+    public interface INClientGenerator
+    {
+        Task<string?> GenerateAsync(string specification, string @namespace, CancellationToken cancellationToken = default);
```

Comment:
In what cases can an empty string be returned?

Reply:
For ex: when templates are empty. Yes, it's rarely case, but possible.

Reply:
I suggest in this case to return an empty string :)

Reply:
ok